crates/loans/src/lib.rs

```diff
@@ -950,22 +997,13 @@ pub mod pallet {
         #[transactional]
         pub fn deposit_all_collateral(origin: OriginFor<T>, asset_id: AssetIdOf<T>) -> DispatchResultWithPostInfo {
             let who = ensure_signed(origin)?;
-            let lend_tokens = Self::free_lend_tokens(asset_id, &who)?;
-            ensure!(!lend_tokens.is_zero(), Error::<T>::InvalidAmount);
-            Self::do_deposit_collateral(&who, lend_tokens.currency(), lend_tokens.amount())?;
-            Ok(().into())
-        }
-
-        #[pallet::weight(<T as Config>::WeightInfo::collateral_asset())]
-        #[transactional]
-        pub fn deposit_collateral(
-            origin: OriginFor<T>,
-            asset_id: AssetIdOf<T>,
-            #[pallet::compact] amount: BalanceOf<T>,
-        ) -> DispatchResultWithPostInfo {
-            let who = ensure_signed(origin)?;
-            ensure!(!amount.is_zero(), Error::<T>::InvalidAmount);
-            Self::do_deposit_collateral(&who, asset_id, amount)?;
+            let free_lend_tokens = Self::free_lend_tokens(asset_id, &who)?;
+            ensure!(!free_lend_tokens.is_zero(), Error::<T>::InvalidAmount);
+            let reserved_lend_tokens = Self::reserved_lend_tokens(asset_id, &who)?;
+            // This check could fail if `withdraw_all_collateral()` leaves leftover lend_tokens locked.
+            // However the current implementation is guaranteed to withdraw everything.
+            ensure!(reserved_lend_tokens.is_zero(), Error::<T>::TokensAlreadyLocked);
+            Self::do_deposit_collateral(&who, free_lend_tokens.currency(), free_lend_tokens.amount())?;
```

Comment:
Let's refactor `do_deposit_collateral` to `do_deposit_all_collateral`, and move the logic above this to inside that function? Since we can't call `do_deposit_collateral` with arbitrary amounts anymore

Reply:
We would end up with a `do_deposit_all_collateral` and a `do_withdraw_collateral`, because the latter is still needed on partial redeems. I'm not sure how much `do_deposit_all_collateral` adds for sacrificing the current uniform interface (where `do_deposit_collateral()` and `do_withdraw_collateral()` are clearly opposites).